src/status_im/common/enter_seed_phrase/view.cljs

```diff
@@ -227,10 +230,12 @@
 
 (defn view
   []
-  (let [{:keys [on-success]} (rf/sub [:get-screen-params])]
+  (let [{:keys [on-success on-unmount]} (rf/sub [:get-screen-params])]
```

Comment:
This looks like a smell @Parveshdhull. Fundamentally we shouldn't couple the event and subscription layers with the components lifecycles directly. The subs are a bit more tied, but not really, because re-frame does the heavy lifting based on reactions.

I don't know if what I'm going to suggest is a good fit, but sometimes, instead of passing opaque callbacks all around, we can pass open/transparent data, which the consumer can contextualize and decide what to do. The tradeoff of this choice might be that the code may not be as general (since a callback can do absolutely anything) but the code can end up being easier to inspect/debug.

What are your thoughts on this pattern @ulisesmac @clauxx? Do you see a better way?

Reply:
Are you suggesting that, instead of passing `on-unmount`, I pass the `onboarding-flow?` parameter and then call `(rf/dispatch [:onboarding/clear-navigated-to-enter-seed-phrase-from-screen])` on `on-unmount` when it is true?

Reply:
Or alternatively, I can cleanup call `on-unmount` and cleanup this state while opening this screen?

Reply:
Another solution, instead of passing on-unmount or onboarding-flow?, is to create an additional wrapper view specifically for onboarding. This approach aligns with the DX task - https://www.notion.so/Explicit-is-better-than-implicit-1a78f96fb65c80e4967bfb43f2e09280

Reply:
@ilmotta I think your suggestion is great, and also what @Parveshdhull said about passing `onboarding-flow?` as param solves it.

I also don't like to pass callbacks because of their opaqueness, although sometimes it's needed.

Reply:
Thank you, @ilmotta and @ulisesmac for the discussion. 

For now, I have migrated to the `onboarding-flow?` parameter. The wrapper approach is still a bit immature, and we need to agree on and create guidelines for naming wrappers, or perhaps consider using a different namespace, etc. I think it would be worth revisiting this later; otherwise, our codebase may end up looking inconsistent with a few wrappers.

Reply:
Nice discussion in this thread, thank you @Parveshdhull. In terms of inconsistency in the repo, there's a benefit in experimenting ideas in develop, learning from developer feedback if it's good, and only later on devising the full guidelines. We did that iteration style a lot with testing utilities. Otherwise we may get in a paralysis situation because the repository is very large and thus reaching full consistency can be challenging. Perhaps in the beginning a light guideline can be defined (?) and the experiment could kick in?